lib/src/main/kotlin/com/wire/integrations/jvm/WireEventsHandler.kt

```diff
@@ -24,29 +24,33 @@ import com.wire.integrations.jvm.service.WireApplicationManager
 import org.slf4j.LoggerFactory
 
 /**
- * Abstract class exposed by the SDK, clients can override the methods in this class and pass it
- * during SDK initialization to handle Wire events.
+ * Abstract class exposed by the SDK to handle events. This subclasses cannot be subclassed
+ * directly, you should subclass [WireEventsHandlerDefault] or [WireEventsHandlerSuspending].
  */
-@Suppress("TooManyFunctions")
-abstract class WireEventsHandler {
-    private val logger = LoggerFactory.getLogger(this::class.java)
-
+sealed class WireEventsHandler {
     /**
      * The [WireApplicationManager] is used to manage the Wire application lifecycle and
      * communication with the backend.
      * NOTE: Do not use manager in the constructor of this class, as it will be null at that time.
      * Use it only inside the event handling methods.
      */
-    protected val manager: WireApplicationManager by lazy {
+    val manager: WireApplicationManager by lazy {
         IsolatedKoinContext.koinApp.koin.get<WireApplicationManager>()
     }
+}
 
-    open fun onNewMessage(wireMessage: WireMessage.Text) {
-        logger.info("Received event: onNewMessage")
-    }
+/**
+ * Create a subclass overriding the methods in this class, and pass it
+ * during SDK initialization to handle Wire events.
+ *
+ * Default version, blocking functions for Java or non-suspending Kotlin code.
+ */
+@Suppress("TooManyFunctions")
+abstract class WireEventsHandlerDefault : WireEventsHandler() {
```

Comment:
Would it be possible to have separate files for `WireEventsHandlerDefault` and `WireEventsHandlerSuspending` as they will for sure continue to grow?